design/Events_design.md

```diff
@@ -0,0 +1,113 @@
+# Events in Obscuro - Design document
+
+## Scope
+
+This is a design proposal for how to handle events in Obscuro.
+
+## What are events in Ethereum
+
+### Smart contracts can emit events which are streamed from a node and generally used by UIs.
+
+For example:
+
+```emit Transfer(from, to, amount);```
+
+
+### A web app can subscribe to events by doing something like:
+
+```
+var subscription = web3.eth.subscribe('logs', {
+    address: '0x123456..', 
+    topics: ['0x12345...']
+}, function(error, result){
+    if (!error)
+        console.log(result);
+});
+- address - String|Array: An address or a list of addresses to only get logs from particular account(s).
+- topics - Array: An array of values which must each appear in the log entries. The order is important, if you want to leave topics out use null, e.g. [null, '0x00...']. You can also pass another array for each topic with options for that topic e.g. [null, ['option1', 'option2']]
+```
+
+A web app might request:
+- stream all "transfer" events where the 'to' field is my address
+- stream all "transfer" events from the USDC contract
+
+
+### How events work 
+
+The query made on the UI side is transformed in a server-side query on the node, and there is some logic after a tx is executed and events are emmitted, to match them against the filters requested by users and distribute them to the requester.(? - need to confirm this is 100% how it works) .
```

Comment:
Can you clarify in the doc whether the events are discarded after being distributed, or are stored for future subscribers? e.g. Can I connect to an Ethereum node and use a subscription to get a list of all events going back to the first block?